cmd/booster-http/http_test.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"testing"
+
+	mocks_booster_http "github.com/filecoin-project/boost/cmd/booster-http/mocks"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewHttpServer(t *testing.T) {
+
+	// Create a new mock Http server
+	ctrl := gomock.NewController(t)
+	httpServer := NewHttpServer("", 7777, false, mocks_booster_http.NewMockHttpServerApi(ctrl))
+	httpServer.Start(context.Background())
+
+	// Check that server is up
+	resp, err := http.Get("http://localhost:7777/")
+	require.NoError(t, err)
+	require.Equal(t, 200, resp.StatusCode)
+
+	// Stop the server
+	err = httpServer.Stop()
+	require.NoError(t, err)
+}
+
+func TestHttpGzipResponse(t *testing.T) {
+
+	// Create a new mock Http server with custom functions
+	ctrl := gomock.NewController(t)
+	httpServer := NewHttpServer("", 7777, false, mocks_booster_http.NewMockHttpServerApi(ctrl))
+	httpServer.ctx, httpServer.cancel = context.WithCancel(context.Background())
+	listenAddr := fmt.Sprintf(":%d", httpServer.port)
+	handler := http.NewServeMux()
+	handler.HandleFunc(httpServer.pieceBasePath(), testHandlePieceRequest)
+	handler.HandleFunc("/", httpServer.handleIndex)
+	handler.HandleFunc("/index.html", httpServer.handleIndex)
+	httpServer.server = &http.Server{
+		Addr:    listenAddr,
+		Handler: handler,
+		// This context will be the parent of the context associated with all
+		// incoming requests
+		BaseContext: func(listener net.Listener) context.Context {
+			return httpServer.ctx
+		},
+	}
+
+	go func() {
+		if err := httpServer.server.ListenAndServe(); err != http.ErrServerClosed {
+			log.Fatalf("http.ListenAndServe(): %w", err)
+		}
+	}()
```

Comment:
The idea of using the mock is that you shouldn't need to copy all the startup code. You can just call httpServer.Start()

Then in the mock, add an EXPECT() for the UnsealSectorAt method that returns a reader over a file, similar to the code here:
https://github.com/filecoin-project/boost/pull/573/files#diff-45d2a3a8bb169fc4f527befb3677309c5bcb522cc8830a8bd88344714b9793d9R96

You may also need to mock out a couple of other calls to make everything work.